classes/migration/upgrade/v3_4_0/I6782_RemovePlugins.inc.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I6782_RemovePlugins.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I6782_RemovePlugins
+ * @brief Remove the usageStats and views report plugin.
+ *
+ * This script has to be called after I6782_Metrics, i.e. after usageStats plugin settings were successfully migrated.
+ */
+
+namespace APP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+use PKP\core\Core;
+use PKP\file\FileManager;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+
+class I6782_RemovePlugins extends Migration
+{
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        $fileManager = new FileManager();
+
+        // Remove usageStats plugin
+        $category = 'generic';
+        $productName = 'usageStats';
+        $pluginVersionsEntryExists = DB::table('versions')
+            ->where('current', 1)
+            ->where('product_type', '=', 'plugins.' . $category)
+            ->where('product', '=', $productName)
+            ->exists();
+        if ($pluginVersionsEntryExists) {
+            $pluginDest = Core::getBaseDir() . DIRECTORY_SEPARATOR . 'plugins' . DIRECTORY_SEPARATOR . $category . DIRECTORY_SEPARATOR . $productName;
+            $pluginLibDest = Core::getBaseDir() . DIRECTORY_SEPARATOR . PKP_LIB_PATH . DIRECTORY_SEPARATOR . 'plugins' . DIRECTORY_SEPARATOR . $category . DIRECTORY_SEPARATOR . $productName;
+            // Delete files
+            // Can we check permissions in the perflight spript?
```

Comment:
I don't think this will actually be necessary at all; when someone upgrades, they should not unpack the new OJS package over top of the old one, so the `usageStats` directories should already not be present after the code is upgraded.